apollo-federation/src/sources/connect/expand/mod.rs

```diff
@@ -0,0 +1,385 @@
+                &mut schema,
+                ComponentName {
+                    origin: ComponentOrigin::Definition,
+                    name: name!("Query"),
+                },
+            )?;
+
+            Ok(schema)
+        }
+
+        fn expand_object(
+            &self,
+            to_schema: &mut FederationSchema,
+            object: &ObjectFieldDefinitionPosition,
+            connector: &Connector,
+        ) -> Result<(), FederationError> {
+            let field_type = object.get(self.original_schema.schema())?;
+
+            // We first need to create the type pointed at by the current field
+            let output_name = field_type.ty.inner_named_type().clone();
+            let output_type = self.original_schema.get_type(output_name)?;
+            let extended_output_type = output_type.get(self.original_schema.schema())?;
+
+            // If the type is built-in, then there isn't anything that we need to do for the output type
+            if !extended_output_type.is_built_in() {
+                match output_type {
+                    TypeDefinitionPosition::Object(out) => {
+                        // Extract the needed fields from the selection
+                        let JSONSelection::Named(ref selections) = connector.selection else {
+                            unimplemented!()
+                        };
+                        let mut field_selections = IndexMap::new();
+                        for selection in selections.selections_iter() {
+                            let as_name = Name::new(selection.name())?;
+                            let field = out
+                                .field(as_name.clone())
+                                .get(self.original_schema.schema())?;
+                            let field = FieldDefinition {
+                                description: field.description.clone(),
+                                name: field.name.clone(),
+                                arguments: field.arguments.clone(),
+                                ty: field.ty.clone(),
+                                directives: ast::DirectiveList::new(),
+                            };
+
+                            field_selections.insert(as_name, Component::new(field));
+                        }
+
+                        let out_def = out.get(self.original_schema.schema())?;
+                        let out_type = ObjectType {
+                            description: out_def.description.clone(),
+                            name: out_def.name.clone(),
+                            implements_interfaces: out_def.implements_interfaces.clone(),
+                            directives: DirectiveList::new(),
+                            fields: field_selections,
+                        };
+
+                        out.pre_insert(to_schema)?;
+                        out.insert(to_schema, Node::new(out_type))?
+                    }
+                    o => todo!("not yet done for {o:?}"),
```

Comment:
could probably go ahead and add branches for scalar and enum types since there's nothing to do there, right?

Reply:
We'll still need to add the scalars / enums to the subgraph, so I'll do that in the next PR!